detection_rules/rule_validators.py

```diff
@@ -235,63 +278,103 @@ def validate_stack_combos(self, data: QueryRuleData, meta: RuleMeta) -> Union[EQ
                 if exc:
                     raise exc
 
-    def validate_integration(self, data: QueryRuleData, meta: RuleMeta, package_integrations: List[dict]) -> Union[
-            EQL_ERROR_TYPES, None, ValueError]:
+    def validate_integration(self, data: QueryRuleData, meta: RuleMeta,
+                             package_integrations: List[dict]) -> Union[EQL_ERROR_TYPES, None, ValueError]:
         """Validate an EQL query while checking TOMLRule against integration schemas."""
         if meta.query_schema_validation is False or meta.maturity == "deprecated":
             # syntax only, which is done via self.ast
             return
 
         error_fields = {}
-        current_stack_version = ""
-        combined_schema = {}
-        for integration_schema_data in get_integration_schema_data(data, meta, package_integrations):
-            ecs_version = integration_schema_data['ecs_version']
-            integration = integration_schema_data['integration']
-            package = integration_schema_data['package']
-            package_version = integration_schema_data['package_version']
-            integration_schema = integration_schema_data['schema']
-            stack_version = integration_schema_data['stack_version']
-
-            if stack_version != current_stack_version:
-                # reset the combined schema for each stack version
-                current_stack_version = stack_version
-                combined_schema = {}
+        package_schemas = {}
+
+        # Initialize package_schemas with a nested structure
+        for integration_data in package_integrations:
+            package = integration_data["package"]
+            integration = integration_data["integration"]
+            if integration:
+                package_schemas.setdefault(package, {}).setdefault(integration, {})
+            else:
+                package_schemas.setdefault(package, {})
+
+        # Process each integration schema
+        for integration_schema_data in get_integration_schema_data(
+            data, meta, package_integrations
+        ):
+            ecs_version = integration_schema_data["ecs_version"]
+            package, integration = (
+                integration_schema_data["package"],
+                integration_schema_data["integration"],
+            )
+            package_version = integration_schema_data["package_version"]
```

Comment:
```suggestion
            package_version = integration_schema_data["package_version"]
```

We are exhausting every ECS version to check that each field is found in each stack version the rule backports to. However, for integration schema validation, we are only validating against the latest. Should we validate every compatible integration package back to the earliest package of the compatible stack version?